tasks/task_manager.hh

```diff
@@ -172,13 +170,16 @@ public:
                     std::function<std::optional<Res>(const task_essentials&)> map_finished_children) const {
                 auto shared_holder = co_await _lock.hold_read_lock();
 
-                co_return boost::join(
-                            _children | boost::adaptors::map_values | boost::adaptors::transformed(map_children),
-                            _finished_children | boost::adaptors::transformed(map_finished_children)
-                        )
-                        | std::views::filter([] (const auto& task) { return bool(task); })
-                        | std::views::transform([] (const auto& res) { return res.value(); })
-                        | std::ranges::to<std::vector<Res>>();
+                auto deopt = std::views::filter([] (const auto& task) { return bool(task); })
+                        | std::views::transform([] (const auto& res) { return res.value(); });
+
+                auto kids = _children | std::views::values | std::views::transform(map_children) | deopt;
+                auto finished_kids = _finished_children | std::views::transform(map_finished_children) | deopt;
+                std::vector<Res> result;
```

Comment:
Reserve `kids.size() + finished_kids.size()`

Reply:
Due to std::views::filter, the size is unknown.

The original code also did not reserve, for the same reason. So the patch is neutral in this regard.

If we want to optimize, we could reserve _children.size() + _finished_children.size() (accepting some wasted space), or we could calculate std::ranges::distance(kids) + std::ranges::distance(finished_kids) (leading to double-scan of the input). But if we optimize it should be separate from removing the boost dependency.